### DISCUSSION TITLE
Error running a simple shiny app in quarto

### DISCUSSION THREAD

**Discussion by ManuelSpinola** [quarto-dev/quarto-cli#963] [Q&A] [answered]
I am trying to learn shiny in quarto, but I tried this very simple app and got me the following error.

---
title: "Hello, world!"
format: html
server: shiny
---

```{r}
#| include: false
data("cars")
d1 <- cars
d2 <- cars
```

```{r}
#| panel: sidebar
selectInput("dataset", label = "Dataset", choices = c("d1", "d2"))
```

```{r}
#| panel: fill
plotOutput('plot1')
```

```{r}
#| context: server

datasetInput <- reactive({
  switch(input$dataset,
         "d1" = d1,
         "d2" = d2)
})

output$plot1 <- renderPlot({
  plot(datasetInput())
})
```
Error: object 'd1' not found

**Comment by cderv** [comment]
Hi @ManuelSpinola , 

Please can you edit your issue and reformat so that it is easier to read and copy ? 
https://yihui.org/issue/#please-format-your-issue-correctly

Thanks!

**Comment by ManuelSpinola** [comment]
Something like that?

````

---
title: "Hello, world!"
format: html
server: shiny
---

```{r}
#| include: false
data("cars")
d1 <- cars
d2 <- cars
```

```{r}
#| panel: sidebar
selectInput("dataset", label = "Dataset", choices = c("d1", "d2"))
```

```{r}
#| panel: fill
plotOutput('plot1')
```

```{r}
#| context: server

datasetInput <- reactive({
  switch(input$dataset,
         "d1" = d1,
         "d2" = d2)
})

output$plot1 <- renderPlot({
  plot(datasetInput())
})
```

Error: object 'd1' not found

````

**Reply by cderv** [reply]
Yes thanks ! You can also edit your first post when you need it like this case.

**Comment by cscheid** [comment, marked as answer]
Notice that you're using the variables d1 and d2 in a code cell with context: server, but defining them in a cell without that metadata. Those cells run in different execution contexts. We explain that [here](https://quarto.org/docs/interactive/shiny/execution.html).
